src/Runner.Worker/Handlers/HandlerFactory.cs

```diff
@@ -84,6 +84,45 @@ public sealed class HandlerFactory : RunnerService, IHandlerFactory
                     }
                     nodeData.NodeVersion = "node16";
                 }
+
+                var localForceActionsToNode20 = StringUtil.ConvertToBoolean(Environment.GetEnvironmentVariable(Constants.Variables.Agent.ManualForceActionsToNode20));
```

Comment:
Wondering why we do the following two checks during `ExecutionContext.InitializeJob` and also here:
- `Environment.GetEnvironmentVariable(Constants.Variables.Agent.ManualForceActionsToNode20)`
- `Global.Variables.GetBoolean("DistributedTask.ForceGithubJavascriptActionsToNode20")`

Seems like having the logic either in InitializeJob or here would be sufficient. Do we need both?

Also, here we do an additionally check and prefer the workflow-level `env`, but we don't check that during `InitializeJob`. Should we add that to `InitializeJob`?

Reply:
I reverted changes in `ExecutionContext`. We don't need them. So we need this logic only here for customer's actions